src/core/linter-rules/check-charset.js

```diff
@@ -29,22 +26,22 @@ const getCharset = element => {
 function linterFunction(conf, doc) {
   const metas = doc.querySelectorAll("meta[charset]");
   const val = [];
-  for (let i = 0; i < metas.length; i++) {
-    val.push(metas[i].outerHTML);
+  for (const meta of metas) {
+    val.push(meta.getAttribute('charset').trim().toLowerCase());
   }
-  const offendingElements = val.filter(getCharset);
+  const utfExists = val.indexOf('utf-8');
 
   //only a single meta[charset] and is set to utf-8, correct case
-  if (!offendingElements.length && metas.length === 1) {
+  if (utfExists !== -1 && metas.length === 1) {
     return [];
   }
   //if more than one meta[charset] tag defined along with utf-8
   //or
   //no meta[charset] present in the document
   return {
     name,
-    offendingElements,
-    occurrences: offendingElements.length,
+    utf_exists: (utfExists !== -1) ? 1: 0,
+    metasLength: metas.length,
```

Comment:
`occurrences` should be kept. We don't need `utf_exists` so we can just skip it. Then only `name`, `occurrences` and `...meta` will remain.